requirements.txt

```diff
@@ -1,7 +1,7 @@
 lit==15.0.6
 # numpy 1.24 deprecates np.object, np.bool, np.float, np.complex, np.str,
 # and np.int which are used heavily in onnx-mlir.
-numpy>=1.19.5, <=1.23.5
+numpy>=1.17.4, <=1.23.5
 protobuf==3.20.3
-pytest>=4.6.9, <=7.2.0
-pytest-xdist>=1.31.0, <=2.5
+pytest>=7.2.0
```

Comment:
What would your thoughts be on something like ([compatible releases](https://peps.python.org/pep-0440/#compatible-release))?
```
pytest~=7.2
pytest-xdist~=3.0
```

The `~` should let us pull in latest version of the 7.2.* release. So we'll pull in the latest security and bug fixes to that version but shouldn't pull in incompatible versions.

Overall I don't like caping because of the security implications. If tomorrow there were a 7.2.3 that fixes a critical CVE, I'd rather pull that in automatically than have a requirements.txt that _forces_ someone to use a _known_ bad version until we go in and approve and push a PR to use the fixed version.

There is the risk that binary provider or build system is poisoned to produce bad packages. However if that's the case the poisoner could also target older so a cap doesn't protect us. Something like Solarwinds was over a possible 14 month period so it's not really possible to "wait out" bad packages.